packages/@tinacms/graphql/src/resolve.ts

```diff
@@ -272,16 +272,38 @@ export const resolve = async ({
                     return { node: document }
                   }),
                 }
-                // TODO when jeffs back: Look at this to make sure its OK to do this. (I am pretty sure it is -- Logan)
-                // Fixes https://github.com/tinacms/tinacms/issues/2886
               } else if (
                 info.fieldName === 'documents' &&
                 value?.collection &&
                 value?.hasDocuments
               ) {
-                // use the collecion and hasDocuments to resolve the documents
+                let filter = args.filter
```

Comment:
@jeffsee55  Mind taking a look at this code? The issue is that `resolveCollectionConnection` expects filter to have this shape.

```ts
{
    [fieldName]: <FilterValue>
}
```

But when doing 
```gql 
{
   collection( name: "posts") {
        documents(filter: {posts: { [fieldName]: <FilterValue>}}}){
        # ....
        }
   }
 }
```


filter has the shape of

```ts
{
[collectionnName]: {
    [fieldName]: <FilterValue>
}
}

```